codecov.yml

```diff
@@ -10,6 +10,6 @@ coverage:
       default:
         # Require 1% coverage, i.e., always succeed
         target: 1
-    patch: false
+    patch: true
```

Comment:
Might be worth doing something like this to explicitly set the target, and maybe a couple other options?
```
    patch:
      default:
        target: 95%
        if_no_uploads: error
        if_not_found: success
        if_ci_failed: failure
```
The default target is `auto`, which based on [the docs](https://docs.codecov.io/docs/commit-status#target) uses the base commit. So maybe that's already okay.

Reply:
Feel free to push that in--I'm a little hesitant to set actual target thresholds and fancier stuff before we have just the basic reporting working.

Reply:
I think we've been pretty annoyed by CI fails in the past when the threshold wasn't right/too high. 

I think that's part of why there's a 1 % in there for the project. A lot of PRs can't reach the coverage for X reasons--they touch build code, or whatever.. And then one can argue that we should ignore coverage of those files or have separate reports, etc..

Reply:
I think we should either explicitly disable it (set to 1% here) or enable it (90%?) and then fix the `ignores` and similar things as we see them pop up -- the way it's currently structured it's not clear to me which we'll actually end up doing.

My vote would be to try the latter and disable it again (set to 1%) if need be.

Reply:
Ok, I've revised based on your feedback.

Note that I don't think I saw the new `patch` codecov entry in our CI list in the first iteration of this PR, so that's probably something to keep an eye on when reviewing after the CI runs again.

Reply:
I think we may also need `require_ci_to_pass: false` somewhere in here, based on a similar discussion in https://github.com/numpy/numpy/pull/16458

Reply:
@thomasrockhu Any ideas why the `patch` diff is not reported in the CI list here--everything is passing now too.

Reply:
Taking a look now

Reply:
Looks like the `yaml` is invalid, left a suggestion that should fix the issue